packages/postcss-unique-selectors/src/index.js

```diff
@@ -3,23 +3,45 @@ const selectorParser = require('postcss-selector-parser');
 
 /**
  * @param {string} selectors
- * @param {selectorParser.SyncProcessor<void>} callback
  * @return {string}
  */
-function parseSelectors(selectors, callback) {
-  return selectorParser(callback).processSync(selectors);
-}
+function generateUniqueSelector(selectors) {
+  /** @type {Map<string, string>} */
+  const uniqueSelectors = new Map();
 
-/**
- * @param {import('postcss').Rule} rule
- * @return {string}
- */
-function unique(rule) {
-  const selector = [...new Set(rule.selectors)];
-  selector.sort();
-  return selector.join();
-}
+  /** @type {selectorParser.SyncProcessor<void>} */
+  const collectUniqueSelectors = (selNode) => {
+    for (const node of selNode.nodes) {
+      /** @type {string[]} */
+      const comments = [];
+
+      // Duplicates are removed by stripping the comments and using the results as the Map key.
+      const keyNode = node.clone();
+      keyNode.walk((sel) => {
+        if (sel.type === 'comment') {
+          comments.push(sel.value);
+          sel.remove();
+        }
+      });
+      const key = keyNode.toString().trim();
+
+      const dupeSelector = uniqueSelectors.get(key);
+      if (!dupeSelector) {
+        uniqueSelectors.set(key, node.toString());
+      } else if (comments.length) {
+        // If the duplicate selector has a comment, it is concatenated to the end of the selector.
+        uniqueSelectors.set(key, `${dupeSelector}${comments.join('')}`);
+      }
+    }
+  };
+
+  selectorParser(collectUniqueSelectors).processSync(selectors);
 
+  return [...uniqueSelectors.entries()]
+    .sort(([a], [b]) => (a > b ? 1 : a < b ? -1 : 0))
```

Comment:
Just a curiosity, does the order of selectors in a selector list have any influence ever?

Reply:
I'm not sure about the need for sorting, but the previous logic seemed to sort so I wanted to respect that 😅 

https://github.com/cssnano/cssnano/pull/1619/files#diff-f56d82ff8f062ba22cac4f017caf4acefe55ce2a2b2ef2267b455a471c0951f0L19